src/unix/linux.c

```diff
@@ -499,8 +516,37 @@ int uv__platform_loop_init(uv_loop_t* loop) {
     munmap(sqe, sqelen);
 
   uv__close(ringfd);
+}
+
+
+static void uv__iou_delete(struct uv__iou* iou) {
+  if (iou->ringfd != -1) {
+    munmap(iou->sq, iou->maxlen);
+    munmap(iou->sqe, iou->sqelen);
+    uv__close(iou->ringfd);
+    iou->ringfd = -1;
+  }
+}
+
+
+int uv__platform_loop_init(uv_loop_t* loop) {
+  uv__loop_internal_fields_t* lfields;
+
+  lfields = uv__get_internal_fields(loop);
+  lfields->ctl.ringfd = -1;
+  lfields->iou.ringfd = -1;
 
-  return 0;  /* Not an error, falls back to thread pool. */
+  loop->inotify_watchers = NULL;
+  loop->inotify_fd = -1;
+  loop->backend_fd = epoll_create1(O_CLOEXEC);
+
+  if (loop->backend_fd == -1)
+    return UV__ERR(errno);
+
+  uv__iou_init(loop->backend_fd, &lfields->iou, 64, UV__IORING_SETUP_SQPOLL);
+  uv__iou_init(loop->backend_fd, &lfields->ctl, 256, 0);
```

Comment:
Any rationale for choosing these ring sizes (forgot to ask this in the initial iouring PR)?

Reply:
64 for the sqpoll ring: no real reason except that sqes aren't expected to sit in the queue for long (on account of the kernel polling thread) so it probably doesn't have to be big.

256 for the epoll_ctl ring: I used the watcher_cross_stop test as guidance because it creates a lot of file descriptors.

Reply:
To underline how 64 is probably plenty big: #3980 - would be caught by tests if the ring ever actually filled up. 🤦